test/markets/markets.e2e-spec.ts

```diff
@@ -0,0 +1,66 @@
+import { INestApplication, ValidationPipe } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { MainModule } from '../../src/main.module';
+
+const request = require('supertest');
+
+describe('Markets', () => {
+
+  let app: INestApplication;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [MainModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+
+    // Validation pipeline is require to check validations.
+    app.useGlobalPipes(
+      new ValidationPipe({
+        whitelist: true,
+        transform: true,
+        forbidNonWhitelisted: true,
+        transformOptions: {
+          enableImplicitConversion: true,
+        },
+      }),
+    );
+    await app.init();
+  });
+
+  it(`GET /markets`, () => {
+    return request(app.getHttpServer()).get('/markets')
```

Comment:
have a look at DTFS set up. we can have a separate config for `supertest` calls.

Reply:
Copied api.js from DTFS. Because of app initialisation differences api.js is loaded not at the top of test file.